executor/aggregate.go

```diff
@@ -860,38 +863,37 @@ func (e *StreamAggExec) consumeGroupRows() error {
 	return nil
 }
 
-func (e *StreamAggExec) fetchChildIfNecessary(ctx context.Context, chk *chunk.Chunk) error {
+func (e *StreamAggExec) fetchChildIfNecessary(ctx context.Context, chk *chunk.Chunk) (err error) {
 	if e.inputRow != e.inputIter.End() {
 		return nil
 	}
 
 	// Before fetching a new batch of input, we should consume the last group.
 	if e.newAggFuncs != nil {
-		err := e.consumeGroupRows()
+		err = e.consumeGroupRows()
 		if err != nil {
 			return errors.Trace(err)
 		}
 	}
 
-	err := e.children[0].Next(ctx, e.childrenResults[0])
+	err = e.children[0].Next(ctx, e.childrenResults[0])
 	if err != nil {
 		return errors.Trace(err)
 	}
+
 	// No more data.
 	if e.childrenResults[0].NumRows() == 0 {
-		if e.hasData || len(e.GroupByItems) == 0 {
-			err := e.appendResult2Chunk(chk)
-			if err != nil {
-				return errors.Trace(err)
-			}
+		if !e.isChildReturnEmpty {
+			err = e.appendResult2Chunk(chk)
```

Comment:
Why don't check the value of `err` here? If you have any particular reason to do so, is it better you add it here?

Reply:
err is checked in line 892

Reply:
got it.